pkg/operation/botanist/controlplane/kubeapiserver/deployment/secret.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployment
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (k *kubeAPIServer) deploySecretOIDCBundle(ctx context.Context) (*string, error) {
+	var (
+		secretOIDCBundle = k.emptySecret(secretNameOIDC)
+		buf              = new(bytes.Buffer)
+		sha256Hex        string
+	)
+
+	// create a copy to create SHA sum - do not create SHA based on current resource that includes the resource version
+	secretOIDCSHA := *secretOIDCBundle
```

Comment:
I don't get it. `secretOIDCBundle` is an empty secret, why can't you directly use it? Also, if you need a copy, use `DeepCopy()` function.